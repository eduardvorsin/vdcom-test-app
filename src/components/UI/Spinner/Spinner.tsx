import {
  Box,
  CircularProgress,
  SxProps,
  Theme,
  styled,
} from '@mui/material';
import React, { FC } from 'react';

const SpinnerWrapper = styled(Box)(() => ({
  display: 'block',
  position: 'absolute',
  top: '50%',
  left: '50%',
  transform: 'translate(-50%,-50%)',
}));

type SpinnerProps = {
  className?: string,
  sx?: SxProps<Theme>,
  size?: number,
}

const Spinner: FC<SpinnerProps> = ({
  className,
  sx,
  size = 60,
}) => (
  <SpinnerWrapper
    className={className}
    sx={sx}
    data-testid='spinner'
  >
    <CircularProgress
      size={size}
    />
  </SpinnerWrapper>
);

export default Spinner;
