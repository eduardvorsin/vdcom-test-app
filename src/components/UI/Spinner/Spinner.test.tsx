import { render, screen } from '@testing-library/react';
import React from 'react';
import Spinner from './Spinner';

describe('Spinner tests', () => {
  it('should render correctly', () => {
    render(
      <Spinner
        size={50}
        className='spinner'
      />,
    );

    expect(screen.getByRole<HTMLSpanElement>('progressbar')).toBeInTheDocument();
  });

  it('should be 100 px high if the prop size value is 100', () => {
    render(
      <Spinner
        size={100}
      />,
    );

    expect(screen.getByRole<HTMLSpanElement>('progressbar')).toHaveStyle({
      width: '100px',
      height: '100px',
    });
  });

  it('is snapshot with default size', () => {
    render(
      <Spinner />,
    );

    expect(screen.getByTestId<HTMLDivElement>('spinner')).toMatchSnapshot();
  });

  it('is snapshot with custom size', () => {
    render(
      <Spinner
        size={150}
      />,
    );

    expect(screen.getByTestId<HTMLDivElement>('spinner')).toMatchSnapshot();
  });
});
