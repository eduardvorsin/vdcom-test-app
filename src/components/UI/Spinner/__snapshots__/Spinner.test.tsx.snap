// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner tests is snapshot with custom size 1`] = `
<div
  class="MuiBox-root css-9xybgt"
  data-testid="spinner"
>
  <span
    class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
    role="progressbar"
    style="width: 150px; height: 150px;"
  >
    <svg
      class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </span>
</div>
`;

exports[`Spinner tests is snapshot with default size 1`] = `
<div
  class="MuiBox-root css-9xybgt"
  data-testid="spinner"
>
  <span
    class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
    role="progressbar"
    style="width: 60px; height: 60px;"
  >
    <svg
      class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </span>
</div>
`;
